packages/now-cli/src/util/dev/server.ts

```diff
@@ -1503,16 +1503,15 @@ export default class DevServer {
     if (!match) {
       // If the dev command is started, then proxy to it
       if (this.devProcessPort) {
-        debug('Proxying to frontend dev server');
+        const upstream = `http://localhost:${this.devProcessPort}`;
+        debug(`Proxying to frontend dev server: ${upstream}`);
         this.setResponseHeaders(res, nowRequestId);
-        return proxyPass(
-          req,
-          res,
-          `http://localhost:${this.devProcessPort}`,
-          this,
-          nowRequestId,
-          false
-        );
+        const origUrl = url.parse(req.url || '/', true);
+        //delete origUrl.search;
```

Comment:
How come this one is commented out? Let's delete if it's problematic?

Reply:
I wasn't sure the purpose of this part. I noticed the search string is deleted in the build match here:

https://github.com/vercel/vercel/blob/5035fa537f82181649d42778c60145d8890f3291/packages/now-cli/src/util/dev/server.ts#L1535-L1539

Should we delete for the dev proxy too?

Reply:
I think the logic should be consistent. I think it is required so that `url.format()` respects the `query` object instead of the `search` string (which would be outdated if there are `uri_args` defined).

Reply:
Confirmed it's necessary:

```
> var u = url.parse('/foo?bar=1', true);
> u.query.a='b'
> url.format(u)
'/foo?bar=1'
> delete u.search
> url.format(u)
'/foo?bar=1&a=b'
```

Reply:
```suggestion
        delete origUrl.search;
```

Reply:
We should probably add a test for this behavior as well.

Reply:
Nice find!